src/commands/goto.ts

```diff
@@ -50,6 +59,70 @@ function executeGoto(gotoType: number, editor: vscode.TextEditor, count: number,
   }
 }
 
+function executeGotoLine(editor: vscode.TextEditor, count: number, extend: boolean, position: string = 'default') {
+  const positions: { [index: string]: {(x: vscode.Selection): number} } = {
+    'first': function (x: vscode.Selection) {
+      return editor.document.lineAt(x.active.line).firstNonWhitespaceCharacterIndex
+    },
+    'end': function (x: vscode.Selection) {
+      return editor.document.lineAt(x.active.line).range.end.character
+    },
+    'start': function (x: vscode.Selection) {
+      return 0
+    },
+    'default': function (x: vscode.Selection) {
+      return 0
+    }
+  }
+  
+  editor.selections = editor.selections.map(x =>
+    {
+      const npos:vscode.Position = new vscode.Position(x.active.line,  (positions[position] || positions['default'])(x));
+      return new vscode.Selection(extend ? x.anchor : npos, npos)
+    }
+  )
+}
+
+function executeGotoDisplayLine(editor: vscode.TextEditor, count: number, extend: boolean, position: string) {
+  const positions: { [index:string]: {(): number} } = {
```

Comment:
Same as `executeGotoLine` with `position`'s type and `positions`'s explicit type annotation.